File: app/src/main/java/ru/tpu/russiantpu/main/fragments/profile/TpuPortalFragment.java
package ru.tpu.russiantpu.main.fragments.profile;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import ru.tpu.russiantpu.R;

/**
 * Фрагмент хранит кнопки,
 * ссылающие на соответствующие разделы портала ТПУ.
 */
public class TpuPortalFragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        final View layoutInflater = inflater.inflate(R.layout.fragment_tpu_portal, container, false);

        layoutInflater.findViewById(R.id.button_personal_card).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                goToLink("private");
            }
        });

        layoutInflater.findViewById(R.id.button_grades).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                goToLink("study");
            }
        });

        layoutInflater.findViewById(R.id.button_scholarship).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                goToLink("stipend");
            }
        });

        layoutInflater.findViewById(R.id.button_payment).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                goToLink("oplata");
            }
        });

        layoutInflater.findViewById(R.id.button_my_payments).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                goToLink("oplata/mypayment");
            }
        });

        return layoutInflater;
    }

    private void goToLink(String url) {
        String baseUrl = "https://portal.tpu.ru/desktop/student/";
        url = baseUrl + url;
        try {
            requireActivity().startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(url)));
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

/*    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.button_personal_card:
                goToLink("private");
                break;
            case R.id.button_grades:
                goToLink("study");
                break;
            case R.id.button_scholarship:
                goToLink("stipend");
                break;
            case R.id.button_payment:
                goToLink("oplata");
                break;
            case R.id.button_my_payments:
                goToLink("oplata/mypayment");
                break;
        }
    }*/
}

File: app/src/main/java/ru/tpu/russiantpu/main/fragments/profile/TpuPortalFragment.kt
package ru.tpu.russiantpu.main.fragments.profile

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import ru.tpu.russiantpu.R
import ru.tpu.russiantpu.utility.PersonalInfoContent

/**
 * Фрагмент хранит кнопки,
 * ссылающие на соответствующие разделы портала ТПУ.
 */
class TpuPortalFragment : Fragment(R.layout.fragment_tpu_portal), PersonalInfoContent {

    private var academicPlanUrl: String? = null
    private var scheduleUrl: String? = null

    override fun onAcademicPlanUrlReceived(url: String) {
        academicPlanUrl = url
    }

    override fun onScheduleUrlReceived(url: String) {
        scheduleUrl = url
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        view.findViewById<View>(R.id.button_academic_plan).setOnClickListener { academicPlanUrl?.let { goToLink(it) } }
        view.findViewById<View>(R.id.button_schedule).setOnClickListener { scheduleUrl?.let { goToLink(it) } }
        view.findViewById<View>(R.id.button_personal_card).setOnClickListener { goToPortalLink("private") }
        view.findViewById<View>(R.id.button_grades).setOnClickListener { goToPortalLink("study") }
        view.findViewById<View>(R.id.button_scholarship).setOnClickListener { goToPortalLink("stipend") }
        view.findViewById<View>(R.id.button_payment).setOnClickListener { goToPortalLink("oplata") }
        view.findViewById<View>(R.id.button_my_payments).setOnClickListener { goToPortalLink("oplata/mypayment") }
        view.findViewById<View>(R.id.button_mail).setOnClickListener { goToLink("https://mail.tpu.ru/") }
    }

    // Переход на одну из страниц портала ТПУ
    private fun goToPortalLink(url: String) {
        val fullUrl = "https://portal.tpu.ru/desktop/student/$url"
        try {
            activity?.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(fullUrl)))
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    // Переход на любой url
    private fun goToLink(fullUrl: String) {
        try {
            activity?.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(fullUrl)))
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}